### DISCUSSION TITLE
How to match multiple lines?

### DISCUSSION THREAD

**Discussion by tennox** [biomejs/gritql#582] [Q&A] [answered]
```gritql
`const location = useLocation()
const navigate = useNavigate()`
=> `const {locnav, location, navigate = useLocationNavigate()`
```
https://app.grit.io/studio?key=zy5VO6zfCYJOlmtXMXiiM

can't get this to work. Tried:
- `const location = useLocation()\nnavigate = useNavigate()`
- `const location = useLocation(); const navigate = useNavigate()`
- `const location = useLocation()\nconst navigate = useNavigate()`

**Comment by tennox** [comment]
I found `sequential` patterns in docs, but doesn't work like I've tried:
```
sequential {
    `const location = useLocation()`,
    `const navigate = useNavigate()`
}
```
or
```
sequential {
    bubble file($body) where $body <: contains `const location = useLocation()`,
    bubble file($body) where $body <: contains `const navigate = useNavigate()`
}
```
https://app.grit.io/studio?key=NYbygZ9vVtcIEnAH3vOOh

**Comment by pavi2410** [comment]
I have figured this out - https://app.grit.io/studio?key=GtxxyFFPPU6CfqHs1Ck1B

![image](https://github.com/user-attachments/assets/d70c0570-5a35-485a-a4d0-43ad29e2b753)

**Comment by pavi2410** [comment, marked as answer]
I also figured how to preserve other enclosed statements in the function scope. https://app.grit.io/studio?key=lri--5R9WfN75ywlpLmQp

![image](https://github.com/user-attachments/assets/164e173b-ff55-4b87-84c3-a98fc40a6e6c)
